### PR TITLE
Lua updates

### PR DESCRIPTION
**Pull request by tpimh**
- Lua 5.1.5 (as `lua51`)
- Lua 5.2.4 (as `lua`, default)
- Lua 5.3.6 (as `lua53`)
- Lua 5.4.6 (as `lua54`)

### CODE REVIEW

**Comment by tpimh** [comment]
Tested with [this code](https://github.com/tpimh/luatest-psp/) on PPSSPP (will test on real hardware later):

- `make LUAV=51`
Screenshot from 2024-06-24 12-50-11
- `make`
Screenshot from 2024-06-24 12-49-49
- `make LUAV=53`
Screenshot from 2024-06-24 12-49-18
- `make LUAV=54`
Screenshot from 2024-06-24 12-48-52

File: lua51/PSPBUILD
Comment (sharkwouter) [inline lua51/PSPBUILD:27]: What does the `a` do?

Comment (tpimh) [reply lua51/PSPBUILD:27]: It's a target that only builds `liblua.a` (on in case of patched makefile, `liblua${LUAV}.a`), all other platforms use `all` target instead which includes `lua` and `luac` executables which are not needed on PSP.

Comment (tpimh) [reply lua51/PSPBUILD:27]: Specifically, it's defined here: [makefile#L65](https://github.com/lua/lua/blob/4324904b60db5243ede68d0922c1bf3c0dd05986/makefile#L65)

Comment (sharkwouter) [reply lua51/PSPBUILD:27]: Makes sense, I was worried it might be a typo

**Review by sharkwouter — CHANGES_REQUESTED** [review]
This looks good, but there is a conflict which has to be resolved.

**Comment by sharkwouter** [comment]
That would be doable. Adding the following lines to all the lua packages should work:
```
provides=('lua')
conflicts=('lua')
```
Make sure to remove the `groups` variable from all but the main one. I would call the main one just lua.

**Comment by tpimh** [comment]
@sharkwouter Hmm... After reading your comment, I am not sure if I did it correctly. I listed all the conflicting packages in `conflicts` and just `lua` in `provides`.

**Comment by sharkwouter** [comment]
Yeah, I would keep one that is named lua. I think it would be good to take a quick look here: https://wiki.archlinux.org/title/PKGBUILD#provides

There it states that it's good to add a version to the provides entry.

**Comment by tpimh** [comment]
Yes, adding a version is good idea.

Also ArchWiki has an entry on [Virtual packages](https://wiki.archlinux.org/title/Pacman#Virtual_packages), so I guess this makes `lua` a virtual package (does not exist by itself, but is provided by multiple other packages).

If we don't want this, I suggest always having the latest version (currently, 5.4) being named just `lua`.

File: lua52/PSPBUILD
Comment (sharkwouter) [inline lua52/PSPBUILD:8]: This should probably be moved to lua54 if we want to make that the default.

Comment (tpimh) [reply lua52/PSPBUILD]: Still needs to be tested how it behaves with several conflicting packages.

Comment (sharkwouter) [reply lua52/PSPBUILD]: I'll merge this to my repo's master, then we can do a test.

Comment (sharkwouter) [reply lua52/PSPBUILD]: Here is the GitHub Action: https://github.com/sharkwouter/psp-packages/actions/runs/9662990777

Once it's complete, you can change the server in `$PSPDEV/etc/pacman.conf` to `https://sharkwouter.github.io/psp-packages/` for testing.

**Comment by sharkwouter** [comment]
If you already have a version of lua installed, it will just update that package when you run `psp-pacman -S lua`.

**Comment by sharkwouter** [comment]
This is what happens if you try to install a conflicting version:
Screenshot from 2024-06-25 15-25-38

**Comment by sharkwouter** [comment]
@tpimh can you take a look at this PR?: https://github.com/pspdev/pspdev/pull/82

We should probably merge that before we merge this change. I'm happy with this solution, though. I'll merge this after we're sure that pspdev can work with it.

**Pull request merged**
